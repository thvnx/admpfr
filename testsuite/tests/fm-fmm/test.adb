with Ada.Text_IO; use Ada.Text_IO;
with Admpfr;      use Admpfr;

procedure Test is
   M, N, O, P, Q : Mpfloat;
begin
   N.Set ("1.1");
   O.Set ("2.2");
   P.Set ("3.3");
   Q.Set ("4.4");

   M.Fma (N, O, P);
   Put_Line (M'Image);

   M.Fms (N, O, P);
   Put_Line (M'Image);

   M.Fmma (N, O, P, Q);
   Put_Line (M'Image);

   M.Fmms (N, O, P, Q);
   Put_Line (M'Image);

end Test;
