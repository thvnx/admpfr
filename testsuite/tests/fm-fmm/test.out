5.7200000000000006e+00
-8.7999999999999945e-01
1.6940000000000001e+01
-1.2100000000000000e+01
